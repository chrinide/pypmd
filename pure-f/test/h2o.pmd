verbose
load h2o.wfn
grid
  lebedev 5810
endgrid
surface 1
