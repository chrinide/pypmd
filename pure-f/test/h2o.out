#INFO: **** input file is /home/jluis/src/git/todo/pmd-0/h2o.py ****
#!/usr/bin/env python

import os, numpy
from pyscf import gto, scf
from pyscf.tools import wfn_format
from pyscf.dft import numint

name = 'h2o'

mol = gto.Mole()
mol.atom = '''
  O  0.0000  0.0000  0.1173
  H  0.0000  0.7572 -0.4692
  H  0.0000 -0.7572 -0.4692
'''
dirnow = os.path.realpath(os.path.join(__file__, '..'))
basfile = os.path.join(dirnow, 'sqzp.dat')
mol.basis = basfile
mol.verbose = 4
mol.spin = 0
mol.symmetry = 1
mol.charge = 0
mol.build()

mf = scf.RHF(mol)
mf.max_cycle = 150
mf.chkfile = name+'.chk'
#mf.__dict__.update(lib.chkfile.load(name+'.chk', 'scf'))
#dm = mf.make_rdm1()
mf.kernel()
dm = mf.make_rdm1()

coeff = mf.mo_coeff[:,mf.mo_occ>0]
occ = mf.mo_occ[mf.mo_occ>0]
energy = mf.mo_energy[mf.mo_occ>0]
wfn_file = name + '.wfn'
with open(wfn_file, 'w') as f2:
    wfn_format.write_mo(f2, mol, coeff, mo_occ=occ, mo_energy=energy)

coords = numpy.random.random((5,3))*3.0
print "Coordinates and density values"
ao_value = numint.eval_ao(mol, coords, deriv=1)
# The first row of rho is electron density, the rest three rows are electron
# density gradients which are needed for GGA functional
rho = numint.eval_rho(mol, ao_value, dm, xctype='GGA')
print coords 
print rho
#INFO: ******************** input file end ********************


System: ('Linux', 'silicio', '3.16.0-4-amd64', '#1 SMP Debian 3.16.51-3 (2017-12-13)', 'x86_64', '')  Threads 12
Python 2.7.14 (default, Apr  9 2018, 16:34:43) 
[GCC 7.3.0]
numpy 1.15.1  scipy 1.1.0
Date: Sun Sep 30 23:08:43 2018
PySCF version 1.5.3
PySCF path  /home/jluis/src/pyscf/pyscf
GIT ORIG_HEAD ce7571e12d7379a6c6b76fb8db4f1eda034a60c7
GIT HEAD      ref: refs/heads/master
GIT master branch  54aca46961fb873f970e377a1f44e0a522e43f00

[ENV] PYSCF_TMPDIR /scratch-ssd/jluis
[CONFIG] conf_file None
[INPUT] verbose = 4
[INPUT] num. atoms = 3
[INPUT] num. electrons = 10
[INPUT] charge = 0
[INPUT] spin (= nelec alpha-beta = 2S) = 0
[INPUT] symmetry 1 subgroup None
[INPUT] Mole.unit = angstrom
[INPUT]  1 O      0.000000000000   0.000000000000   0.117300000000 AA    0.000000000000   0.000000000000   0.221664874411 Bohr
[INPUT]  2 H      0.757200000000   0.000000000000  -0.469200000000 AA    1.430900621521   0.000000000000  -0.886659497646 Bohr
[INPUT]  3 H     -0.757200000000   0.000000000000  -0.469200000000 AA   -1.430900621521   0.000000000000  -0.886659497646 Bohr

nuclear repulsion = 9.1895337629349
point group symmetry = C2v
num. orbitals of irrep A1 = 64
num. orbitals of irrep A2 = 27
num. orbitals of irrep B1 = 50
num. orbitals of irrep B2 = 36
number of shells = 51
number of NR pGTOs = 220
number of NR cGTOs = 177
basis = /home/jluis/src/git/todo/pmd-0/sqzp.dat
ecp = {}
CPU time:         0.93


******** <class 'pyscf.scf.hf_symm.SymAdaptedRHF'> flags ********
method = SymAdaptedRHF-RHF
initial guess = minao
damping factor = 0
level shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
DIIS start cycle = 1
DIIS space = 8
SCF tol = 1e-09
SCF gradient tol = None
max. SCF cycles = 150
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = h2o.chk
max_memory 64000 MB (current use 67 MB)
Freeze 0 electrons in irreps []
    10 free electrons in irreps A1 A2 B1 B2
Set gradient conv threshold to 3.16228e-05
init E= -75.8836731086601
HOMO (B2) = -0.479641335460967  LUMO (A1) = 0.0141095000335925
cycle= 1 E= -76.0274940856938  delta_E= -0.144  |g|= 0.442  |ddm|= 0.766
HOMO (B2) = -0.435474832328485  LUMO (A1) = 0.025018152133593
cycle= 2 E= -76.0530730897037  delta_E= -0.0256  |g|= 0.263  |ddm|= 0.269
HOMO (B2) = -0.51816103166776  LUMO (A1) = 0.0261616640858065
cycle= 3 E= -76.0635649923819  delta_E= -0.0105  |g|= 0.0286  |ddm|= 0.0835
HOMO (B2) = -0.509485425782872  LUMO (A1) = 0.0270674726040808
cycle= 4 E= -76.0637363150061  delta_E= -0.000171  |g|= 0.00605  |ddm|= 0.0113
HOMO (B2) = -0.510357020406656  LUMO (A1) = 0.0270895454751747
cycle= 5 E= -76.0637510935483  delta_E= -1.48e-05  |g|= 0.00109  |ddm|= 0.00432
HOMO (B2) = -0.510542931572288  LUMO (A1) = 0.0270830744935426
cycle= 6 E= -76.0637518452383  delta_E= -7.52e-07  |g|= 0.000204  |ddm|= 0.00102
HOMO (B2) = -0.510607636238259  LUMO (A1) = 0.0270834726540049
cycle= 7 E= -76.0637518744307  delta_E= -2.92e-08  |g|= 3.48e-05  |ddm|= 0.00026
HOMO (B2) = -0.510609742927472  LUMO (A1) = 0.0270838532417893
cycle= 8 E= -76.0637518751045  delta_E= -6.74e-10  |g|= 3.6e-06  |ddm|= 3.59e-05
HOMO (B2) = -0.510609887636212  LUMO (A1) = 0.0270839394717799
Extra cycle  E= -76.0637518751109  delta_E= -6.34e-12  |g|= 1.26e-06  |ddm|= 7.11e-06
converged SCF energy = -76.0637518751109
Coordinates and density values
[[0.68868146 1.06321449 2.83441324]
 [1.55197281 2.08674961 2.43132285]
 [2.25463456 0.89543877 2.33476173]
 [0.65720763 0.58347429 2.28145371]
 [0.31609876 1.66559717 1.19095168]]
[[ 0.00293366  0.00088928  0.00112233  0.01509917  0.03549166]
 [-0.00190992 -0.00103936 -0.00205152 -0.01261518 -0.01635665]
 [-0.00244445 -0.00115738 -0.00064604 -0.00929691 -0.07615529]
 [-0.00653053 -0.00136933 -0.00178094 -0.03533467 -0.04640767]]
